src/main/java/uk/gov/digital/ho/hocs/info/api/TeamResource.java

```diff
@@ -159,16 +158,15 @@ public ResponseEntity deleteTeamPermissions(@PathVariable String teamUUID, @Requ
 
         for (UserDto user: users){
             Set<Team> teams = teamService.getTeamsForUser(UUID.fromString(user.getId()));
-            ArrayList<UUID> teamUUIDS = new ArrayList<>();
+            ArrayList<String> teamNames = new ArrayList<>();
+            Map<String, String> unitNames = new HashMap<>();
```

Comment:
Should this be Map<String, List<String>>?

Then you would do unitNames.hasKey("UnitName") and if it exists just append to the list with the team. Else you create a List with the team and add it to the map?

Reply:
You could then get rid of the teamNames list?